images/virt-artifact/patches/040-set-reboot-policy.patch

```diff
@@ -0,0 +1,219 @@
+diff --git a/pkg/virt-launcher/notify-client/client.go b/pkg/virt-launcher/notify-client/client.go
+index 5da5f68103..6e236964bf 100644
+--- a/pkg/virt-launcher/notify-client/client.go
++++ b/pkg/virt-launcher/notify-client/client.go
+@@ -3,6 +3,7 @@ package eventsclient
+ import (
+ 	"context"
+ 	"fmt"
++	"os"
+ 	"path/filepath"
+ 	"sync"
+ 	"time"
+@@ -528,6 +529,50 @@ func (n *Notifier) StartDomainNotifier(
+ 		}
+ 	}
+ 
++	domainQemuMonitorEventShutdownCallback := func(c *libvirt.Connect, d *libvirt.Domain, event *libvirt.DomainQemuMonitorEvent) {
++
++		type Result struct {
++			Event   string `json:"event"`
++			Details string `json:"details"`
++		}
++
++		log.Log.Infof("Domain Qemu Monitor Shutdown event received")
++
++		name, err := d.GetName()
++		if err != nil {
++			log.Log.Reason(err).Info(cantDetermineLibvirtDomainName)
++		}
++
++		f, err := os.OpenFile("/dev/termination-log", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
++		if err != nil {
++			log.Log.Reason(err).Error("Failed to open /dev/termination-log")
++			return
++		}
++		defer f.Close()
++
++		r := Result{
++			Event:   event.Event,
++			Details: event.Details,
++		}
++
++		b, err := json.Marshal(r)
++		if err != nil {
++			log.Log.Reason(err).Error("Failed to marshal result to JSON")
++			return
++		}
++		b = append(b, '\n')
++
++		if _, err = f.Write(b); err != nil {
++			log.Log.Reason(err).Error("Failed to write to termination-log")
++		}
++
++		select {
++		case eventChan <- libvirtEvent{Domain: name}:
++		default:
++			log.Log.Infof(libvirtEventChannelFull)
++		}
++	}
++
+ 	err := domainConn.DomainEventLifecycleRegister(domainEventLifecycleCallback)
+ 	if err != nil {
+ 		log.Log.Reason(err).Errorf("failed to register event callback with libvirt")
+@@ -567,6 +612,16 @@ func (n *Notifier) StartDomainNotifier(
+ 		log.Log.Reason(err).Errorf("failed to register event callback with libvirt")
+ 		return err
+ 	}
++	events := []string{
++		"SHUTDOWN", "RESET",
```

Comment:
Do we need a RESET event? We use guest-reset reason from the Detail.Reason of the SHUTDOWN event.